### PR TITLE
Fixes a OSD bug when un-/muting and adds a toggle to display volume as 0 when muting

### PR DESCRIPTION
**Pull request by painerp**
Since I borked the last PR (#134) I thought I would take the time to explain it more properly now.

This fixes a bug that the displayed values didn't match the actual volume of the affected device, but instead seemed to be stuck on a previously measured value of eg. brightness or something else unrelated.

For example:
- Set microphone volume to 50%
- Increase brightness to 100%
- Mute microphone

The osd icon shows the muted microphone but the label and level bar show 100%, which is unrelated to the microphone.

\
This also implements a toggle to show the volume as "0" when muting the device in the on screen display.

Hope this clarifies it a bit more and thanks for the input @Jas-SinghFSU

### CODE REVIEW

**Comment by Jas-SinghFSU** [comment]
@painerp Works awesomely! Once you resolve the merge conflicts we should be good to merge.

**Comment by painerp** [comment]
@Jas-SinghFSU Thanks, I resolved the conflicts and also removed the overflow class from brightness changes so they don't overlap anymore.

File: modules/osd/bar/index.ts [hunk 1]
```diff
@@ -13,19 +14,29 @@ export const OSDBar = (ort: OSDOrientation) => {
...
                     self.hook(brightness, () => {
+                        self.class_names = self.class_names.filter(c => c !== "overflow");
                         self.value = brightness.kbd;
                     }, "notify::kbd")
                     self.hook(audio.microphone, () => {
                         self.toggleClassName("overflow", audio.microphone.volume > 1)
                         self.value = audio.microphone.volume <= 1 ? audio.microphone.volume : audio.microphone.volume - 1;
                     }, "notify::volume")
+                    self.hook(audio.microphone, () => {
+                        self.toggleClassName("overflow", audio.microphone.volume > 1 && (!options.theme.osd.muted_zero.value || !(audio.microphone.is_muted !== false)));
```
Comment (Jas-SinghFSU) [inline modules/osd/bar/index.ts:29]:
```suggestion
                        self.toggleClassName("overflow", audio.microphone.volume > 1 && (!options.theme.osd.muted_zero.value || audio.microphone.is_muted === false));
```
We can simplify this logic so in the scenario that muted_zero IS enabled, overflow should only happen if `is_muted` is false (statement yields true). If it's null or true then that evaluates to false which won't execute the overflow. We don't have to double negate here.

File: modules/osd/bar/index.ts [hunk 2]
```diff
@@ -13,19 +14,29 @@ export const OSDBar = (ort: OSDOrientation) => {
...
+                        self.value = (options.theme.osd.muted_zero.value && audio.microphone.is_muted !== false) ? 0 : audio.microphone.volume <= 1 ? audio.microphone.volume : audio.microphone.volume - 1;
+                    }, "notify::is-muted")
                     self.hook(audio.speaker, () => {
                         self.toggleClassName("overflow", audio.speaker.volume > 1)
                         self.value = audio.speaker.volume <= 1 ? audio.speaker.volume : audio.speaker.volume - 1;
                     }, "notify::volume")
+                    self.hook(audio.speaker, () => {
+                        self.toggleClassName("overflow", audio.speaker.volume > 1 && (!options.theme.osd.muted_zero.value || !(audio.speaker.is_muted !== false)));
```
Comment (Jas-SinghFSU) [inline modules/osd/bar/index.ts:37]:
```suggestion
                        self.toggleClassName("overflow", audio.speaker.volume > 1 && (!options.theme.osd.muted_zero.value || audio.speaker.is_muted === false));
```

File: modules/osd/label/index.ts [hunk 3]
```diff
@@ -15,19 +16,29 @@ export const OSDLabel = (ort: OSDOrientation) => {
...
                 self.hook(brightness, () => {
+                    self.class_names = self.class_names.filter(c => c !== "overflow");
                     self.label = `${Math.round(brightness.kbd * 100)}`;
                 }, "notify::kbd")
                 self.hook(audio.microphone, () => {
                     self.toggleClassName("overflow", audio.microphone.volume > 1)
                     self.label = `${Math.round(audio.microphone.volume * 100)}`;
                 }, "notify::volume")
+                self.hook(audio.microphone, () => {
+                    self.toggleClassName("overflow", audio.microphone.volume > 1 && (!options.theme.osd.muted_zero.value || !(audio.microphone.is_muted !== false)));
```
Comment (Jas-SinghFSU) [inline modules/osd/label/index.ts:31]:
```suggestion
                    self.toggleClassName("overflow", audio.microphone.volume > 1 && (!options.theme.osd.muted_zero.value || audio.microphone.is_muted === false));
```

File: modules/osd/label/index.ts [hunk 4]
```diff
@@ -15,19 +16,29 @@ export const OSDLabel = (ort: OSDOrientation) => {
...
+                    self.label = `${options.theme.osd.muted_zero.value && audio.microphone.is_muted !== false ? 0 : Math.round(audio.microphone.volume * 100)}`;
+                }, "notify::is-muted")
                 self.hook(audio.speaker, () => {
                     self.toggleClassName("overflow", audio.speaker.volume > 1)
                     self.label = `${Math.round(audio.speaker.volume * 100)}`;
                 }, "notify::volume")
+                self.hook(audio.speaker, () => {
+                    self.toggleClassName("overflow", audio.speaker.volume > 1 && (!options.theme.osd.muted_zero.value || !(audio.speaker.is_muted !== false)));
```
Comment (Jas-SinghFSU) [inline modules/osd/label/index.ts:39]:
```suggestion
                    self.toggleClassName("overflow", audio.speaker.volume > 1 && (!options.theme.osd.muted_zero.value || audio.speaker.is_muted === false));
```

File: modules/osd/label/index.ts [hunk 5]
```diff
@@ -15,19 +16,29 @@ export const OSDLabel = (ort: OSDOrientation) => {
...
+                    self.class_names = self.class_names.filter(c => c !== "overflow");
                     self.label = `${Math.round(brightness.kbd * 100)}`;
                 }, "notify::kbd")
                 self.hook(audio.microphone, () => {
                     self.toggleClassName("overflow", audio.microphone.volume > 1)
                     self.label = `${Math.round(audio.microphone.volume * 100)}`;
                 }, "notify::volume")
+                self.hook(audio.microphone, () => {
+                    self.toggleClassName("overflow", audio.microphone.volume > 1 && (!options.theme.osd.muted_zero.value || !(audio.microphone.is_muted !== false)));
+                    self.label = `${options.theme.osd.muted_zero.value && audio.microphone.is_muted !== false ? 0 : Math.round(audio.microphone.volume * 100)}`;
```
Comment (Jas-SinghFSU) [inline modules/osd/label/index.ts:32]:
```suggestion
                        self.value = (options.theme.osd.muted_zero.value && audio.microphone.is_muted !== false) ? 0
                            : audio.microphone.volume <= 1 ? audio.microphone.volume
                                : audio.microphone.volume - 1;

```
Let's make this more readable

**Pull request merged**